Compilation error in AliceO2

I have successfully installed the ALFA, FAIRROOT, Aliroot and DDS. During complitation AliceO2 I got the following error.
/home/hushnud/Desktop/My_Alice_O2/O2/Utilities/aliceHLTwrapper/src/aliceHLTEventSampler.cxx: In function ‘int sendSocketPropertiesDDS(std::vector<SocketProperties_t>&)’:
/home/hushnud/Desktop/My_Alice_O2/O2/Utilities/aliceHLTwrapper/src/aliceHLTEventSampler.cxx:520:34: error: no matching function for call to ‘dds::intercom_api::CKeyValue::CKeyValue()’
     dds::intercom_api::CKeyValue ddsKeyValue;
                                  ^
/home/hushnud/Desktop/My_Alice_O2/O2/Utilities/aliceHLTwrapper/src/aliceHLTEventSampler.cxx:520:34: note: candidates are:
In file included from /home/hushnud/Desktop/My_Alice_O2/O2/Utilities/aliceHLTwrapper/src/aliceHLTEventSampler.cxx:36:0:
/home/hushnud/Desktop/My_Alice_O2/ALFA/DDS/include/dds_intercom.h:59:13: note: dds::intercom_api::CKeyValue::CKeyValue(dds::intercom_api::CIntercomService&)
             CKeyValue(CIntercomService& _service);
             ^
/home/hushnud/Desktop/My_Alice_O2/ALFA/DDS/include/dds_intercom.h:59:13: note:   candidate expects 1 argument, 0 provided
/home/hushnud/Desktop/My_Alice_O2/ALFA/DDS/include/dds_intercom.h:46:15: note: constexpr dds::intercom_api::CKeyValue::CKeyValue(const dds::intercom_api::CKeyValue&)
         class CKeyValue
/home/hushnud/Desktop/My_Alice_O2/O2/Utilities/aliceHLTwrapper/src/aliceHLTEventSampler.cxx:549:85: note:   candidate expects 2 arguments, 3 provided
make[2]: *** [Utilities/aliceHLTwrapper/CMakeFiles/aliceHLTEventSampler.dir/src/aliceHLTEventSampler.cxx.o] Error 1
make[1]: *** [Utilities/aliceHLTwrapper/CMakeFiles/aliceHLTEventSampler.dir/all] Error 2
make: *** [all] Error 2

--- a/Utilities/aliceHLTwrapper/src/aliceHLTEventSampler.cxx
+++ b/Utilities/aliceHLTwrapper/src/aliceHLTEventSampler.cxx
@@ -30,6 +30,14 @@
 #define HAVE_FAIRMQ_INTERFACE_CHANGESTATE_STRING
 #endif
 
+// DDS intercom API has been changed in release 1.4
+// as it is not a matter of a few lines, DDS support is disabled for
+// the moment. We need a major rewrite and testing
+#ifdef ENABLE_DDS
+#define ENFORCED_DDS_DISABLE
+#undef ENABLE_DDS
+#endif
+
 #ifdef ENABLE_DDS
 #include <mutex>
 #include <condition_variable>
@@ -271,7 +279,11 @@ int main(int argc, char** argv)
 
   if (bUseDDS) {
 #ifndef ENABLE_DDS
+#ifdef ENFORCED_DDS_DISABLE
+    cerr << "Fatal: device is not compatible with DDS Intercom API any more" << endl;
+#else
     cerr << "Fatal: device has not been compiled with DDS support" << endl;
+#endif
     exit(ENOSYS);
 #endif
     int result=preprocessSocketsDDS(inputSockets, networkPrefix);

--- a/Utilities/aliceHLTwrapper/src/aliceHLTWrapper.cxx
+++ b/Utilities/aliceHLTwrapper/src/aliceHLTWrapper.cxx
@@ -35,6 +35,14 @@
 #define HAVE_FAIRMQ_INTERFACE_CHANGESTATE_STRING
 #endif
 
+// DDS intercom API has been changed in release 1.4
+// as it is not a matter of a few lines, DDS support is disabled for
+// the moment. We need a major rewrite and testing 
+#ifdef ENABLE_DDS
+#define ENFORCED_DDS_DISABLE
+#undef ENABLE_DDS
+#endif
+
 #ifdef ENABLE_DDS
 #include <mutex>
 #include <condition_variable>
@@ -293,7 +301,11 @@ int main(int argc, char** argv)
 
   if (bUseDDS) {
 #ifndef ENABLE_DDS
+#ifdef ENFORCED_DDS_DISABLE
+    cerr << "Fatal: device is not compatible with DDS Intercom API any more" << endl;
+#else
     cerr << "Fatal: device has not been compiled with DDS support" << endl;
+#endif
     exit(ENOSYS);
 #endif
     int result=preprocessSocketsDDS(inputSockets, networkPrefix);

